### PR TITLE
Provide notifications via websockets

### PR DESCRIPTION
**Pull request by justmoon**
Adds a new /accounts/:name/transfers endpoint which provides transfer notifications via the websocket protocol.

### CODE REVIEW

**Comment by justmoon** [comment]
Just looked at the test failures and after thinking about it some more, I noticed an error in thinking - I'm sending the notifications before the database transaction has returned, which is faster, but unsafe. The ledger may end up sending out a notification about a transfer state update that never ends up happening due to a database failure.

Please hold off on reviewing until I can fix this issue.

**Comment by justmoon** [comment]
Ok, ready to review!

File: src/lib/notificationWorker.js [hunk 1]
```diff
@@ -50,18 +53,38 @@ class NotificationWorker {
 
   * queueNotifications (transfer, transaction) {
     const affectedAccounts = _([transfer.debits, transfer.credits])
-      .flatten().pluck('account').map((account) => this.uri.make('account', account)).value()
+      .flatten().pluck('account').value()
     affectedAccounts.push('*')
+
+    const fulfillment = yield this.Fulfillment.findByTransfer(transfer.id, { transaction })
```
Comment (MatthewPhinney) [inline src/lib/notificationWorker.js]: We could save some time by only looking for the fulfillment if the transfer state is `executed` or `rejected`.

**Comment by MatthewPhinney** [comment]
Out of curiosity, do we have any data regarding how this will affect performance?

File: src/lib/notificationWorker.js (hunk 1 above)
Comment (justmoon) [inline src/lib/notificationWorker.js]: Good idea, will add.

**Comment by justmoon** [comment]
@MatthewPhinney 
> Out of curiosity, do we have any data regarding how this will affect performance?

Not yet, but @nguyenripple is working on adding performance tests to CI.

Logically, a websocket frame is a single TCP packet as opposed to an HTTP request which opens up a whole new TCP connection. We are also avoiding at least two database writes (writing the notification intent and then deleting it) by dropping the guaranteed delivery requirement.

Anecdotally, I tried it in five-bells-demo with universal mode and it's visibly faster. Note that universal mode will see a much greater improvement than atomic, since this change speeds up both phases in universal and only the preparation phase in atomic.

Also note that the longer the payment path, the more of an improvement you'll see and in the demo we're dealing usually with two, three or even more hops, whereas our integration tests only go up to two hops currently.

**Comment by clark800** [comment]
LGTM 👍

**Pull request merged**